Assorted (mostly) atmosphere-oriented map changes

atlantiscze opened the pull request:
- This will be quite exhaustive. I've split it into two categories. A changelog entry will be added in separate commit.

Generic changes:
- Most maintenance corridors now have a single vent per atmospheric zone. This allows for refilling from distro, if you don't mind the low speed. No scrubbers are present.
- Few sets of all-access airlocks mapped in primary hallways. This should help contain breaches a bit more. Extra focus has been put on arrivals as this area is very susceptible to breaches due to it's large size.
- Gets rid of few remaining "sandwich" shutter-airlock-shutter installations in primary hallways. All airlocks should now have a shutter under themselves instead of using sandwich method, so the hallways are now standardised with rest of the station.
- Tweaks the arrivals shuttle. It will now get depressurised when the arrivals hallway next to it gets depressurised, as the airlocks to it are forced open (to simulate that the shuttle is docked). Before you say i'm insane this is so the automated alert system can detect an atmosphere problem. Currently, a player joining into arrivals does not get warned, as the shuttle is usually safe, but they are trapped there until someone fixes the breach. After this change, the automatic alert system will warn the player when they try to spawn there, and they will be able to pick other spawn point. Less work for admins - no need to put up the classic "Try to avoid joining via arrivals" announcement, and less trapped players. A win-win situation.
- Maps in an APC into old genetics lab. That ensures the airlocks will work in there.
- Maps in few lights into the hallways, where i noticed dark spots.
- Maps in an autolathe into engineering workshop. Gets rid of useless exosuit fabricator in drone fabrication area (it was currently possible to salvage the drone fab to build an autolathe without needing research. I guess mapping one in is a fair approach)
- Removes two emergency shutters at the holodeck, that were covering the vent and scrubber for the holographic part. This is necessary so this area can be correctly refilled by the vent.
- Minor mapping changes have been made to medbay to make it a bit less of a ZAS trap. Few airlocks and windows have been added to make the area a bit more atmospherically safe. Added airlocks are all-access so they won't prevent people from leaving when treated.
- Adds a camera into detective's office.
- Incinerator has been officially closed. Part of the piping as well as the phoron/oxygen gas cans were removed, doors taped out, and the whole area is abandoned. Still usable for antag shenanigans, with less risk of burning down the station because someone wanted to incinerate a mouse or something.

Engine related changes:
- Few changes have been made to the engine, mostly to the main radiator. I am putting this into separate category for clarity, and i'll write as much details here as possible.
- Original radiator has been removed and replaced by three smaller interconnected radiators mounted on engine room's surface. This has both aesthetic and functional reason. Currently, a single weak meteor that damages a single HE pipe will completely destroy the engine's cooling. After this change, the radiator is much harder to destroy. Meteors will still cause damage, and reduce it's cooling efficiency considerably, but one meteor hit shouldn't mean the engine will be left without cooling completely. This buys a bit more time for engineers to fix this situation.
- Engine room has been expanded by one tile south to make room for a radiator pipe. Few windows have been added around the engine room. For safety purposes, they all have blast doors under them that can be closed.
- Original four coolant canisters removed and replaced by 3x N2, 3x CO2 and 3x PH "engine setup" canisters (the same canister is used by setup-supermatter admin verb). These canisters have double amount of gas in them, so it's like if there were six canisters of each gas there. This should mean that engineers no longer have to break into atmospherics and steal local canisters.
- One N2 canister in engineering hard storage replaced with CO2. Now it's 1x N2, 1x CO2 and 1x PH. Once again, lets people choose what they want for the intercooler without the atmospherics break-in step.
- Some pipes have been moved around to make space for radiator connections. One emergency cooling valve is gone. The other one is in place, and there seems to be absolutely no difference from my observation as valves seem to have limitless transfer rate anyway, so there's no need to have two.
- Engine setup admin verb map markers have been updated accordingly.
- The radiation suits are now placed in front of the radiation signs, rather than behind them (it irked me so much, why would someone put the only protective equipment behind the signs that warn you there's radiation behind them? Also, possible minor visual improvement depending on your preference.
- A note about the radiators: new radiators have 118 HE segments total, the original radiator seems to have 116 HE segments total. They should behave more or less the same.
- Fixes #13726

atlantiscze committed:
Assorted atmosphere-oriented map changes
- This will be quite exhaustive. I've split it into two categories. A changelog entry will be added in separate commit.

Generic changes:
- Most maintenance corridors now have a single vent per atmospheric zone. This allows for refilling from distro, if you don't mind the low speed. No scrubbers are present.
- Few sets of all-access airlocks mapped in primary hallways. This should help contain breaches a bit more. Extra focus has been put on arrivals as this area is very susceptible to breaches due to it's large size.
- Gets rid of few remaining "sandwich" shutter-airlock-shutter installations in primary hallways. All airlocks should now have a shutter under themselves instead of using sandwich method, so the hallways are now standardised with rest of the station.
- Tweaks the arrivals shuttle. It will now get depressurised when the arrivals hallway next to it gets depressurised, as the airlocks to it are forced open (to simulate that the shuttle is docked). Before you say i'm insane this is so the automated alert system can detect an atmosphere problem. Currently, a player joining into arrivals does not get warned, as the shuttle is usually safe, but they are trapped there until someone fixes the breach. After this change, the automatic alert system will warn the player when they try to spawn there, and they will be able to pick other spawn point. Less work for admins - no need to put up the classic "Try to avoid joining via arrivals" announcement, and less trapped players. A win-win situation.
- Maps in few lights into the hallways, where i noticed dark spots.
- Maps in an autolathe into engineering workshop. Gets rid of useless exosuit fabricator in drone fabrication area (it was currently possible to salvage the drone fab to build an autolathe without needing research. I guess mapping one in is a fair approach)
- Removes two emergency shutters at the holodeck, that were covering the vent and scrubber for the holographic part. This is necessary so this area can be correctly refilled by the vent.
- Minor mapping changes have been made to medbay to make it a bit less of a ZAS trap. Few airlocks and windows have been added to make the area a bit more atmospherically safe. Added airlocks are all-access so they won't prevent people from leaving when treated.
- Few cameras have been mapped in around the station to cover various blind spots which i believe should be covered: Detective's office, Incinerator.
- Incinerator should be a little less of a death trap when used incorrectly. It will still be damaged, but shouldn't burn down the whole maintenance. Engineering shuttle has been moved a bit, so being inside in the corner when the engine is active shouldn't result in being irradiated.

Engine related changes:
- Few changes have been made to the engine, mostly to the main radiator. I am putting this into separate category for clarity, and i'll write as much details here as possible.
- Original radiator has been removed and replaced by three smaller interconnected radiators mounted on engine room's surface. This has both aesthetic and functional reason. Currently, a single weak meteor that damages a single HE pipe will completely destroy the engine's cooling. After this change, the radiator is much harder to destroy. Meteors will still cause damage, and reduce it's cooling efficiency considerably, but one meteor hit shouldn't mean the engine will be left without cooling completely. This buys a bit more time for engineers to fix this situation.
- Engine room has been expanded by one tile south to make room for a radiator pipe. Few windows have been added around the engine room. For safety purposes, they all have blast doors under them that can be closed.
- Original four coolant canisters removed and replaced by 3x N2, 3x CO2 and 3x PH "engine setup" canisters (the same canister is used by setup-supermatter admin verb). These canisters have double amount of gas in them, so it's like if there were six canisters of each gas there. This should mean that engineers no longer have to break into atmospherics and steal local canisters.
- One N2 canister in engineering hard storage replaced with CO2. Now it's 1x N2, 1x CO2 and 1x PH. Once again, lets people choose what they want for the intercooler without the atmospherics break-in step.
- Some pipes have been moved around to make space for radiator connections. One emergency cooling valve is gone. The other one is in place, and there seems to be absolutely no difference from my observation as valves seem to have limitless transfer rate anyway, so there's no need to have two.
- Engine setup admin verb map markers have been updated accordingly.
- The radiation suits are now placed in front of the radiation signs, rather than behind them (it irked me so much, why would someone put the only protective equipment behind the signs that warn you there's radiation behind them? Also, possible minor visual improvement depending on your preference.
- A note about the radiators: new radiators have 118 HE segments total, the original radiator seems to have 116 HE segments total. They should behave more or less the same.

atlantiscze committed:
Tweaks unit tests + adds changelog
- APC/scrubber/vent exception list for Exodus updated.

atlantiscze commented:
Screenshots below.

Engine room and adjacent areas:
![image](https://cloud.githubusercontent.com/assets/5147960/16828839/b48501c8-4993-11e6-957b-3eda97ee6919.png)


Arrivals shuttle:
![image](https://cloud.githubusercontent.com/assets/5147960/16828844/c1c08eac-4993-11e6-9b65-23892ef3ee74.png)


Medbay:
![image](https://cloud.githubusercontent.com/assets/5147960/16828850/d19f5b14-4993-11e6-8bec-6d50da9e9d91.png)

PsiOmegaDelta commented:
What's the motivation behind adding camera coverage to the incinerator, one of the few large antag allowing areas we have on the station?

atlantiscze commented:
The camera is on engineering network, to allow engineers to look around the area. There's phoron and oxygen stored there, so i don't think the area should be left completely without any coverage, as these gases (well, mostly phoron) are usually locked up due to the possible danger. Disabling cameras is very simple anyway, and any antag that wishes to do nasty stuff there can do so very quickly - there's only one camera there, and it's easily accessible.

I supose it's not a major point of this PR - if you want me to remove the camera, fine, but i think it's better to have it there.

atlantiscze committed:
Final tweaks
- Applies few remaining map changes - mostly adding vents to maintenance i've missed and which travis notified me about.

Kasuobes commented:
The emergency treatment center (paramedic prep area + sleepers) is supposed to be a ready room for medics to rush out the side entrance, why wall it off with windows and make people go through 2 extra doors?

Airlocks right in front of the psyc office leading to the patient wing and virology are edging the tinted windows. Putting airlocks there also makes the medbay feel way more cramped, not a big fan of it. Ditto for airlocks sandwiched between medical equipment and the CMO's office.

PsiOmegaDelta commented:
The AI will still be able to see the area without restriction, with the noteworthy exception of a disabled camera.
Would rather see the incinerator turned into an abandoned area before losing one of the few antagging zones we have.

atlantiscze commented:
An alternative is current ZASTrap medbay. If one part of medbay gets breached, due to how ZAS works, the whole medbay usually gets vented before emergency shutters react and seal off the breach.

As for the paramedic room, it doesn't sound as something that would be used as a waiting area. From the furniture which i didn't change it looks as another storage. Medical isn't my most played department, but i don't remember anyone ever sitting there waiting to head out. Most people seem to hang out in the front lobby or in the medbay break room to the right.

I'll do something about the incinerator.

atlantiscze committed:
Changes Incinerator
- Incinerator is no longer in working condition from roundstart. Phoron and oxygen, as well as few pipes have been removed. An engineer with pipe dispenser will be able to bring it into working condition within a minute or two relatively easily. The area is taped off, but accessible for various antag shenanigans. Given how horribly incinerator worked, and given that it has pretty much zero actual uses it's probably for the best to keep it like that. If anyone wants to actually use it, the buttons and airlock are still there, just the pipes have to be patched up and gas added via canisters.
- Incinerator camera has been replaced with uninstalled camera assembly laying on the ground.

Kasuobes commented:
Reception and break room are the major hang out areas because they have consoles to watch sensors on, but the prep room is still the ready room with the mech recharger. Maybe add airlocks instead of all windows so emergency responders has a straight path in and out of the prep room and sleepers.

PsiOmegaDelta commented:
The sandwich airlocks were also rather handy in reducing the issues caused by people who don't care about emergency shutters (i.e. basically everyone) spreading atmospheric issues.

atlantiscze commented:
If i remember correctly, anyone can open an emergency shutter which is not blinking. However, if it is blinking (and therefore, leading to dangerous area), it should have more strict requirements (engineering ID i think). The "sandwich" shutters didn't blink, as the other side of shutter was usually blocked by an airlock/other shutter. With these replaced with regular under-the-airlock shutters it should actually get a bit better as people won't be allowed into areas where they may spread atmospheric problems. It will be also easier to identify that there's a problem on the other side, as people will see the flashing lights.

PsiOmegaDelta commented:
Why would the emergency shutters in a sandwich setup that's facing the atmospheric issue not blink, just because they have airlocks/another set of emergency shutters on their other side?
People merely bring crowbars and willful ignorance.

atlantiscze commented:
See screenshot.

On the right, the sandwich version. Notice that the alarm lights on firelocks are not blinking.
On the bottom-left, a shutter-below-airlock version. Notice the lights are blinking correctly.

In the sandwich version, the firelocks warning lights will not blink until you open the airlock in the middle. I think that's because the airlock is taking up the whole tile, and de facto acts as a wall when closed. Once you open the airlock (and wait about 5s, it takes a while to react) the lights finally turn on. A relatively quick player can rush through before the firedoors even have chance to react.

![image](https://cloud.githubusercontent.com/assets/5147960/16829954/4073a43e-499c-11e6-9583-b9cb0c6e610e.png)

PsiOmegaDelta commented:
The firelocks on the left should be flashing. That they are not is a bug.

Techhead0 commented:
Firelocks don't flash from low pressure. They flash from a pressure differential. Because the glass airlocks are closed, the firelocks only have a reading on the west side.

BlueNexus commented:
I strongly disagree with the medbay changes, especially the changes to the paramedic area, for the reasons stated above.

mwerezak commented:
This will conflict with #13676

comma commented:
Medbay layout is quite bad, 'emergency' entrance now doesn't offer any shortcut really, since you have to go around windows that prevent paramedics etc from moving out fast.
Station doesn't need to be ultra super safe, ZAS trap medbay is fine.

Raptor1628 commented:
Medbay changes seem unnecessary, the main hall airlocks were probably kept "sandwich" style exactly so people could move more safely through them. People who ignore firelocks are going to ignore them regardless of flashing in many cases. 

The engineering changes seem fine, though aren't great aesthetically. Closets in windows just look bad, why not move the signs up and keep the lockers where they fit? Also, make the window shutters start closed, safer that way, and makes the engine room feel a lot less open.

Keep the second emergency cooling valve, it is connecting different parts of the loops.

I think part of the point of Genetics not having an APC is that it was broken down, and could be crowbarred.

If the incinerator is "closed," there's less of a chance for people to have an excuse for going in there, and it's easier to see that someone HAS been in there if it's taped off.

BlueNexus commented:
I must point out that this defeats the entire point of the emergency entrance and the paramedic room, which are both supposed to allow rapid access.

atlantiscze committed:
Minor changes according to feedback

atlantiscze commented:
Feedback changes implemented:
- Radiation suits are no longer placed partial windows. I moved the window by one tile which freed a bit of space. Screenshot:

![image](https://cloud.githubusercontent.com/assets/5147960/16887651/ba302c6c-4adb-11e6-9666-189593823268.png)
- Engine radiator shutters are now closed on roundstart (button can still be used to open them)
- Few more minor changes to medbay loadout, mostly to allow faster exit from the EMT storage (i still fail to see why is rapid access to storage needed, but it seems to be the majority's opinion).
- Medbay hallway doors have also been moved a bit to what i believe is a good compromise between nothing as it is now, and what i originally proposed.
- CMO's office has lost one of it's four access doors. On the bright side, that now means the CMO's office can have both a command console and a medical console. Medbay screenshot:

![image](https://cloud.githubusercontent.com/assets/5147960/16887670/d2dfe20c-4adb-11e6-8bc8-82810daa5580.png)


- Incinerator airlock is no longer covered by tape so it's not possible to see whether someone entered the area or not. Furthermore, the airlock is now all-access.
- (Bonus) mapped in few more rechargers for tablet/laptop users to recharge their devices.

atlantiscze committed:
Merge branch 'dev' of https://github.com/Baystation12/Baystation12 into 2016-07-14-MapItIn

OOZ662 commented:
A really, really minor thing that's been bugging me is that the tiny "art storage" room across from the vault has both a tube and a bulb for lighting. Perhaps, given the other minor lighting changes, you'll consider choosing one to keep in there instead of both.

Kasuobes commented:
Another minor request: there are 3 wall turfs that need fixing, would be nice to see them de-uglified.

![](http://i.imgur.com/QiQyNDm.png)
![](http://i.imgur.com/euXXjFx.png)
![](http://i.imgur.com/jj4QIpG.png)

atlantiscze commented:
- Light bulb has been removed from the auxiliary storage
- Walls @Kasuobes pointed out were corrected to be of the same type as adjacent walls.

comma commented:
>i still fail to see why is rapid access to storage needed
it's not about storage, there's emergency exit there. Having supplies being easily reachable form diagnoistics/sleeper room is nice too. I think that another airlock would be nice actually, in line with two existing, storage-paramedics-sleepers, so it's almost as traversable as current version but not as 'ZAS trap'.

Kasuobes commented:
Possible to fix #13726 as well?

The ready room is for keeping emergency equipment so medics can quickly grab supplies (and the mech) and rush out the side door at a moment's notice. Echoing chinsky a door to connect the prep area and the sleepers will be nice, so medics can quickly throw people into sleepers to stabilize them.

atlantiscze commented:
13726 fixed, and requested airlock added.

mwerezak commented:
Please keep genetics without an APC, it was much more atmospheric that way.

atlantiscze commented:
I don't really think the APC makes any notable difference from visual perspective. The change was a quality of life improvement so people can use the airlocks to pass through the area without having to carry a crowbar.

mwerezak commented:
It makes a big difference to atmosphere that the doors and other area machines of the unused laboratory are depowered. Personally I find it strange that you are willing to completely sacrifice that atmosphere for a completely fictional "quality of life improvement," as if not requiring a crowbar to open those doors was better or somehow desirable.

mwerezak commented:
More than that though, I find it strange that you claim not needing a crowbar to access or pass through the area is "an improvement." How is removing the opportunity for gameplay an improvement exactly?

ZCaliber commented:
I, for one, would prefer either the APC to remain unbuilt as it is now, or simply remove the 'genetics' access from the area, in favor of general science and medical access for their respective sides.

atlantiscze commented:
I've went with following compromise:
- The genetics lab APC is no longer built, some of the components may be present for it scattered around the room (random spawner was used)
- Genetics lab was renamed to "Abandoned Lab".
- Lab's airlocks are now powered by whatever room is on the other side. Furthermore, access requirements on these airlocks were reduced (on the medbay side) or completely removed (the maintenance side) so you don't need almost any access to get there.
- Those two small windows which could be used to see into the lab were replaced by walls. Now the area is a dark hole with no camera coverage, ideal for both antag shenanigans, and various construction projects.

atlantiscze committed:
Tweaks to engine waste handling
- Now supports either waste canister storage (current method), or venting waste directly into space. This can be controlled by a digital valve.

atlantiscze commented:
I've added few more changes to the engine waste handling system. It now allows for dumping of waste into space, via a digital valve. Furthermore it allows you to pump gas from cold/hot loop into the waste system.

Original function still remains - there is a canister that can be used to stockpile the waste gas in. The intercooler had to be reduced in size to make space for a dumping vent but it should still be enough for keeping the waste cooled to sufficient levels

![image](https://cloud.githubusercontent.com/assets/5147960/16935292/6b3cf498-4d5e-11e6-90c8-1c85d8508054.png)

Kasuobes commented:
Bumping, status?

atlantiscze commented:
This is pretty much ready feature-wise, just waiting for :+1: from devs.

As for the conflict, i'll resolve that.

atlantiscze committed:
Merge branch 'dev' of https://github.com/Baystation12/Baystation12 into 2016-07-14-MapItIn

mwerezak commented:
- Could you cover the radiator manifolds in grilles, like what was done for the intercooler back when it was introduced? i.e. the black pipe segments in space in this screenshot:

![](https://cloud.githubusercontent.com/assets/5147960/16828839/b48501c8-4993-11e6-957b-3eda97ee6919.png)
- I'm a bit disappointed that the genetics airlocks are now powered, still haven't seen a compelling justification for doing so. I appreciate that the APC has been left unbuilt though.
- Minor criticism of the engine control room locker placement, more for your own benefit when doing future work with maps than anything else:
  ![](https://cloud.githubusercontent.com/assets/5147960/16887651/ba302c6c-4adb-11e6-9666-189593823268.png)
  Having the lockers placed like this is really ugly and illogical. Have both radiation suit closets on one side and both O2 closets on the other. I've seen this in some of your other mapping work as well, I think you would benefit from trying not to overuse symmetry like this. While in this particular case it's really obvious how making the closets symmetrical makes the map look messier and less organized, it's also true in general that a little bit of asymmetry makes a map a lot less bland.
- Those r-glass dividers in the above screenshot are also worth mentioning. Then can be nice when used judiciously but creating narrow 1-tile walkways is really awkward and I think should be avoided unless you are trying to create something that looks strange and alien. In any case having the closets there seems like a poor choice, I don't think there is any arrangement that won't look awkward. I think you are better off putting them somewhere else in that hallway, and extending the control room back to it's original size and shape.
- Please remove the window behind the shuttle thrusters. Again, it's really illogical. I think the best maps are those that look like something that future humans might actually build, and we should strive for that.
  ![](https://cloud.githubusercontent.com/assets/5147960/16828844/c1c08eac-4993-11e6-9b65-23892ef3ee74.png)
- Have you addressed people's concerns about not having direct access to the side exit from the EMT ready room? Incidentally if you brought that back and then added a suit sensor monitoring console plus a place to sit I am sure that EMT players will use it as a place to idle while waiting on-call for emergencies.

![](https://cloud.githubusercontent.com/assets/5147960/16828850/d19f5b14-4993-11e6-8bec-6d50da9e9d91.png)

mwerezak commented:
Overall I still have a negative opinion about the bulk of these changes. If only it was actually about partitioning medbay to handle breaches better.

atlantiscze committed:
Radiator tweaks
- Adds grilles to radiator manifolds
- Moves the radiation/O2 lockers in front of engine control a bit more.
- Both of these changes were suggested by mwerezak.

atlantiscze commented:
@mwerezak:
I've covered the manifolds with grilles as requested. I've also swapped the lockers.

BlueNexus commented:
I also dislike the vast majority of these changes, primarily the medbay changes.

atlantiscze commented:
@mwerezak: Next time please add another comment instead of editing the original one after i've answered to it. I don't get email notifications for edits, while i do get them for new comments, and it helps speed up everything.

First of all, (@BlueNexus may be interested in this part too) the medbay comments were already resolved a long time ago. The side entrance was added, and the airlocks that split medbay were moved a bit to keep the original intent of this PR, while also allowing unobstructed access to the most used area (treatment room). If you want a console added into the EMT storage, it'll be done. I don't have a problem with that. I still think it's a storage, not a meeting room (the door was actually labeled to be a storage), and that majority of EMTs will still prefer to hang out in the lobby with others, but i guess it's okay.

Good idea with the shuttle, i'll replace the window behind thrusters with walls.

I can remove the single pane windows next to lockers too, it's a minor thing. Though, i'm not sure where else to place the lockers to still have them easily accessible, while not being inside potentionally radioactive area. I think the current placement is quite okay.
